extensions/kafka/src/main/java/com/hazelcast/jet/kafka/KafkaDataConnection.java

```diff
@@ -141,7 +144,10 @@ public <K, V> KafkaProducer<K, V> getProducer(@Nullable String transactionalId)
     @Override
     public void destroy() {
         if (producer != null) {
```

Comment:
Cache `producer` in local variable, otherwise if might be true and in the next line the producer may be null.

Also, nullify producer before closing it (close using local variable)

Reply:
maybe simply make `destroy` synchronized as we did in `HazelcastDataConnection`?